src/System.Devices.Gpio/System.Devices.Gpio.Samples/Bme280.cs

```diff
@@ -0,0 +1,419 @@
+        };
+
+        private Bme280_Calibration_Data _cal_data;
+
+        private float _temperature;
+        private float _humidity;
+        private float _pressure;
+        private int _tFine;
+
+        private SpiConnectionSettings _spiSettings;
+        private SpiDevice _spiDevice;
+
+        private Pin _csPin;
+        private Pin _mosiPin;
+        private Pin _misoPin;
+        private Pin _sckPin;
+
+        /// <summary>
+        /// Use this for hardware SPI.
+        /// </summary>
+        public Bme280(Pin cs, SpiConnectionSettings spiSettings)
+        {
+            _csPin = cs;
+            _spiSettings = spiSettings;
+        }
+
+        /// <summary>
+        /// Use this for software SPI.
+        /// </summary>
+        public Bme280(Pin cs, Pin mosi, Pin miso, Pin sck)
+        {
+            _csPin = cs;
+            _mosiPin = mosi;
+            _misoPin = miso;
+            _sckPin = sck;
+        }
+
+        public void Dispose()
+        {
+            if (_spiDevice != null)
+            {
+                _spiDevice.Dispose();
+                _spiDevice = null;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the temperature calibration offset in degrees celsius.
+        /// </summary>
+        public float TemperatureCalibrationOffset { get; set; }
+
+        public void ReadSensor()
+        {
+            ReadTemperature();
+            ReadHumidity();
+            ReadPressure();
+        }
+
+        public float TemperatureInCelsius => _temperature + TemperatureCalibrationOffset;
+
+        public float TemperatureInFahrenheit => TemperatureInCelsius * 1.8F + 32;
+
+        public float Humidity => _humidity;
+
+        public float PressureInPascals => _pressure;
+
+        public float PressureInHectopascals => _pressure / 100;
+
+        public float PressureInMillibars => PressureInHectopascals;
+
+        public bool Begin()
+        {
+            Dispose();
+
+            _csPin.Mode = PinMode.Output;
+            DigitalWrite(_csPin, PinValue.High);
+
+            if (_spiSettings != null)
+            {
+                // Hardware SPI
+                _spiSettings.Mode = SpiMode.Mode0;
+                _spiSettings.ClockFrequency = 500 * 1000;
+                _spiSettings.DataBitLength = 8;
+
+                _spiDevice = new UnixSpiDevice(_spiSettings);
+            }
+            else
+            {
+                // Software SPI
+                _sckPin.Mode = PinMode.Output;
+                _mosiPin.Mode = PinMode.Output;
+                _misoPin.Mode = PinMode.Input;
+            }
+
+            byte chipId = Read8(BME280_REGISTER_CHIPID);
+
+            if (chipId != 0x60)
+            {
+                Console.WriteLine($"chipId = '{chipId:X2}'");
+                return false;
+            }
+
+            ReadSensorCoefficients();
+
+            // Set Humidity oversampling to 1
+            // Set before CONTROL (DS 5.4.3)
```

Comment:
Why do we want to set either of these values this way?  can you add more information?

Reply:
They don't explain to much about it, they just say: "Changes to this register only became effective after a write operation to CONTROL register.". I will add this comment to the source code.